src/datanode/src/tests/test_util.rs

```diff
@@ -50,6 +50,10 @@ impl MockInstance {
     pub(crate) fn inner(&self) -> &Instance {
         &self.instance
     }
+
+    pub(crate) async fn restart(&self) -> Result<()> {
+        self.instance.start().await
```

Comment:
I think we should not call `start()` twice. It'd be better to create a new instance and start it. You might need to hold the `opts` in the MockInstance.
```suggestion
        self.instance = Instance::with_mock_meta_client(&self.opts).await.unwrap();
        instance.start().await.unwrap();
```